Firestore/core/src/model/base_path.h

```diff
@@ -150,8 +150,18 @@ class BasePath {
            std::equal(begin(), end(), potential_child.begin());
   }
 
+  /**
+   * Compares the current path against another Path object. Paths are compared
+   * segment by segment, prioritizing numeric IDs (e.g., "__id123__") in numeric
+   * ascending order, followed by string segments in lexicographical order.
+   */
   util::ComparisonResult CompareTo(const T& rhs) const {
-    return util::CompareContainer(segments_, rhs.segments_);
```

Comment:
CompareContainer is a generic method that compares items in 2 T& variables by iteration, and it is only used here in base_path. The sorting order is unique to base_path(specifically resource_path), makes more sense to add the logic here instead of modifying the CompareContainer function.